conf/modules.config

```diff
@@ -10,8 +10,29 @@
 ----------------------------------------------------------------------------------------
 */
 
-// TODO tie all ext.args to nextflow.config
-// TODO mix in versions of all outputs
+
+def reformat_output(filename_in, outputtype, merge_string, meta_){
+    /*
+        filename_in GString: The file or directory being renamed
+        outputtype GString|null: string specifying the type of output to run, or null if there is not type
+        merge_string Gstring: String to fill in as a tool name
+        meta_: metadat passed in
+    */
+
+    def allowed_compression_types = ["gz"]
+    def output_type = outputtype
+    def file_extension_delimiter = '.'
+
+    def parent = filename_in.tokenize(File.separator)[0..<-1].join(File.separator) ?: null
```

Comment:
Would this also work to get you the parent directory:

```groovy
def parent = Nextflow.file(filename_in).getParent()
```

From https://www.nextflow.io/docs/latest/script.html#getting-file-attributes

Similar to other attributes/parts of the file. Can you check if you can use `Nextflow.file()` here?

Reply:
The variable `filename_in` is not a type that can be easily coerced into a type of file it is of type `ConfigObject` even when calling toString() on the object an error is thrown not allowing me to use file attributes on them. Sadly within the function block I cant even introspect on the object fully to pull out any attributes that could be of use. 

When trying to call file methods I get the following error even after calling `toString()` on the object in I will update the docs string to include this however. 

```
ERROR ~ Unknown method invocation `file` on ConfigObject type -- Did you mean?
    size           
```

Reply:
Thanks for letting me know about this. I was not aware of the difficulty in getting it working.

Testing this out, the error message you get above is saying that the `Nextflow` part of `Nextflow.file` is of type `ConfigObject`. I made a mistake in the syntax and you have to use the full qualified name of the method, that is `nextflow.Nextflow.file()`. Then it will work:

For example, if I add this code:

```groovy
    def filename_in_nf = nextflow.Nextflow.file(filename_in)
    println "filename_in: type=${filename_in.getClass()}, value=${filename_in}; filename_in_nf: type=${filename_in_nf.getClass()}, value=${filename_in_nf}, parent=${filename_in_nf.getParent()}"
```

Then I get the following as output:

```
filename_in: type=class java.lang.String, value=short.json; filename_in_nf: type=class sun.nio.fs.UnixPath, value=./mikrokondo/short.json, parent=./mikrokondo
```

Reply:
That being said, it looks like using `nextflow.Nextflow.file()` is giving the incorrect path since it's giving the path relative to the current directory, whereas the file in the config here seems to just be a file name for me, likely relative to the working directory.

Do you know if this will work properly in Azure? I'm not actually sure how it is working.

Reply:
Looking at my output more, I can see other cases where you have parent directories.

I do think there is a way to use Java's `java.nio.file.Paths.get(filename_in).getParent()`, which won't make the path relative to the current working directory.

That being said, I'm not sure if that always will work. I think the solution you have right now by manipulating strings works and is perfectly fine. Sorry about the long comment.